src/azul/service/storage_service.py

```diff
@@ -32,10 +38,17 @@ def head(self, object_key: str) -> dict:
     def get(self, object_key: str) -> bytes:
         return self.client.get_object(Bucket=self.bucket_name, Key=object_key)['Body'].read()
 
-    def put(self, object_key: str, data: bytes, content_type: Optional[str] = None, **kwargs) -> str:
+    def put(self,
+            object_key: str,
+            data: bytes,
+            content_type: Optional[str] = None,
+            tagging: Optional[Mapping[str, str]] = None,
+            **kwargs) -> str:
         params = {'Bucket': self.bucket_name, 'Key': object_key, 'Body': data, **kwargs}
         if content_type is not None:
             params['ContentType'] = content_type
+        if tagging is not None:
```

Comment:
put() is never called with `tagging` (or your previous `file_name` argument). You should catch these things.